navigation/screens/review-form.tsx

```diff
@@ -34,121 +28,39 @@ export default function TabReviewForm() {
   // description
   const [description, setDescription] = useState("");
 
-  // add photo modal
+  // image uploader and carousel
+  const { addImage } = useImages("reviewForm");
   const [modalVisible, setModalVisible] = useState(false);
-  const ImageUploader = ({
-    isVisible,
-    onClose,
-  }: {
-    isVisible: boolean;
-    onClose: () => void;
-  }) => {
-    return (
-      <Modal
-        animationType="slide"
-        transparent={true}
-        visible={isVisible}
-        onRequestClose={onClose}
-      >
-        <View style={styles.centeredView}>
-          <View style={styles.modalView}>
-            {MainButton("Take Photo", addUsingCamera)}
-            {MainButton("Choose from Gallery", addFromGallery)}
-            {CancelButton("Cancel", () => {
-              onClose();
-            })}
-          </View>
-        </View>
-      </Modal>
-    );
-  };
-
-  // modal to display image
-  const [displayImageVisible, setDisplayImageVisible] = useState(false);
-  const [imageToDisplay, setImageToDisplay] = useState("");
-  const DisplayImage = ({
-    isVisible,
-    onClose,
-  }: {
-    isVisible: boolean;
-    onClose: () => void;
-  }) => {
-    return (
-      <Modal
-        animationType="slide"
-        transparent={true}
-        visible={isVisible}
-        onRequestClose={onClose}
-      >
-        <View style={styles.centeredView}>
-          <View style={styles.modalView}>
-            <Image style={styles.modalImage} source={{ uri: imageToDisplay }} />
-            {CancelButton("Cancel", () => {
-              onClose();
-            })}
-          </View>
-        </View>
-      </Modal>
-    );
-  };
-
-  // images uploaded
-  const [images, setImages] = useState<
-    Array<ImagePicker.ImagePickerSuccessResult>
-  >([]);
-
-  // permissions
-  const [cameraStatus] = ImagePicker.useCameraPermissions();
-  const [galleryAccessStatus] = ImagePicker.useMediaLibraryPermissions();
-
-  // add photo from device gallery
-  const addFromGallery = async () => {
-    if (galleryAccessStatus?.status !== "granted") {
-      const { status } =
+  const handleAddImage = async (source: "camera" | "gallery") => {
```

Comment:
possible let me look it up

Reply:
A bit of a complicated process because you have to hash the image data. Since we are cropping the image each time we add it, there will always be a new img uri so its impossible to compare the uri's without hashing.